packages/data-provider/src/controllers/submissionController.ts

```diff
@@ -58,14 +61,24 @@ const controller = (dependencies: BaseDependencies) => {
 		deleteEntityName: validateRequest(submissionDeleteEntityNameRequestSchema, async (req, res, next) => {
 			try {
 				const submissionId = Number(req.params.submissionId);
-				const entityName = req.params.entityName;
+				const actionType = SUBMISSION_ACTION_TYPE.parse(req.params.actionType.toUpperCase());
+
+				const entityName = req.query.entityName;
+				const index = req.query.index ? parseInt(req.query.index) : undefined;
```

Comment:
@joneubank may disagree, but I think this is a good place to use `Null` as an explicit description of the value being missing "on purpose".
Setting something `= undefined` smells.

Reply:
Thanks for the suggestion. Code fixed.

Reply:
a couple things here...

First, `req.query.index` could be 0, which would be falsey. This should explicitly check if index is undefined. `req.query.index !== undefined ? parseInt(req.query.index) : undefined;`

Second, is this a case where we need to differentiate between "no value provided" and "intentionally has no value". If we need to query the DB for records where the index === null (since DBs do use null values and can filter on those) then including `null` as an option is a good idea.  However, I think this is just a case where the `index` is an optional filter, which is best served by the type `index?: number` in a TS idiomatic way.